napari_animation/frame_sequence.py

```diff
@@ -57,16 +57,20 @@ def _rebuild_frame_index(self, event=None):
         """Create a map of frame number -> (kf0, kf1, fraction)"""
         self._frame_index.clear()
         self._cache.clear()
-        if len(self._key_frames) < 2:
+        if len(self._key_frames) == 0:
             self.events.n_frames(value=len(self))
             return
 
-        f = 0
-        for kf0, kf1 in pairwise(self._key_frames):
-            for s in range(kf1.steps):
-                fraction = s / kf1.steps
-                self._frame_index[f] = (kf0, kf1, fraction)
-                f += 1
+        else:
+            f = 0
+            if len(self._key_frames) == 1:
+                kf1 = self._key_frames[0]
```

Comment:
I don't really understand the purpose of this little section, could you explain it?

Reply:
Previously, if there was only one keyframe, it was not added as a frame and _frames was empty. I just added this here so _frames would be empty only when there are no keyframes at all.